### PR TITLE
feat(core): require set a password

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

If check "set a password" on SIE configs, then users sign in or register through verification code will get an error `user.require_password`, and need to call the new route `/session/sign-in/continue/password` to set a password, then will auto sign in with last verification state.

## Testing

UTs

### CODE REVIEW

File: packages/core/src/routes/session/continue.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        })
+      );
+
+      const { passwordEncrypted, passwordEncryptionMethod } = await encryptUserPassword(password);
+      const updatedUser = await updateUserById(userId, {
+        passwordEncrypted,
+        passwordEncryptionMethod,
+      });
+      const signInExperience = await findDefaultSignInExperience();
+      await checkRequiredProfile(ctx, provider, updatedUser, signInExperience);
```
Comment (darcyYe) [inline packages/core/src/routes/session/continue.ts:44]: not sure whether we still need to do the check again here. Is this reserved for other potentially required user profile fields in the future? But in the current stage, we only concern about whether `password` exists.

File: packages/core/src/routes/session/continue.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      expect(response.statusCode).toEqual(200);
+      expect(updateUserById).toHaveBeenCalledWith(mockUser.id, expect.anything());
+      expect(response.body).toHaveProperty('redirectTo');
+      expect(interactionResult).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({ login: { accountId: mockUser.id } }),
+        expect.anything()
+      );
+    });
```
Comment (darcyYe) [inline packages/core/src/routes/session/continue.test.ts:77]: As this mockUser already has a password, not sure this is valid in a real case.

Comment (darcyYe) [reply packages/core/src/routes/session/continue.test.ts:77]: If password already exists, then checkProfileRequired should not write a continueSignIn payload into session storage.

File: packages/core/src/routes/session/continue.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        password: string().regex(passwordRegEx),
+      }),
+    }),
+    async (ctx, next) => {
+      const { password } = ctx.guard.body;
+      const { userId } = await getContinueSignInResult(ctx, provider);
+      const user = await findUserById(userId);
+
+      assertThat(
+        !user.passwordEncrypted && Object.keys(user.identities).length === 0,
```
Comment (darcyYe) [inline packages/core/src/routes/session/continue.ts:32]: `Object.keys(user.identities).length === 0`
how does this relate to social identities?

**Comment by darcyYe** [comment]
Take a look on UTs.

File: packages/core/src/routes/session/continue.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/session/continue.ts:44]: For the future

File: packages/core/src/routes/session/continue.ts (hunk 3 above)
Comment (wangsijie) [reply packages/core/src/routes/session/continue.ts:32]: social identities can replace password.

File: packages/core/src/routes/session/continue.test.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/session/continue.test.ts]: Yeah... updated.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/core/src/routes/session/continue.ts (hunk 3 above)
Comment (darcyYe) [reply packages/core/src/routes/session/continue.ts:33]: Double check if this case happens when we choose None for sie-v2 sign-up?

**Review by darcyYe — APPROVED** [review]
LGTM

**Pull request merged**